shared-bindings/index.rst

```diff
@@ -21,48 +21,11 @@ Modules
 
 Support Matrix
 ---------------
-NOTE 1: **All Supported** means the following ports are supported: SAMD21, SAMD21 Express,
-SAMD51, SAMD51 Express, and ESP8266.
-
-NOTE 2: **SAMD** and/or **SAMD Express** without additional numbers, means both SAMD21 & SAMD51 versions
-are supported.
-
-NOTE 3: The `pIRkey SAMD21 board <https://www.adafruit.com/product/3364>`_ is specialized and may not
-have modules as listed below.
 
 =================  ==============================
-Module             Supported Ports
+Module             Not Available On
```

Comment:
I'm not sure I like the negation of this because it's now unpleasant.

Regardless of the negation, the listing would be simplified by combining a set of boards into a standard group that could be listed below the table.

Reply:
>I'm not sure I like the negation of this because it's now unpleasant.

I do not disagree at all.

>the listing would be simplified by combining a set of boards into a standard group that could be listed below the table.

I remembered that Dan had mentioned several moons ago that he wanted to get away from a Cartesian type map, as well as move towards a exclusion based system. Its just difficult with rST/Jinja to do non-table-Cartesian (for me, at least).

Grouping shouldn't be difficult; this is already reading `mpconfigboard.mk` for info.

A couple ideas that come to mind:

1. Add Port grouping columns. Minimizes adding to vertical space/scrolling.
```
===========  ===========================================
Module       Not Available On
             -------------------------------------------
             SAMD21 (M0)                   | SAMD51 (M4)
--------------------------------------------------------
audioio      feather_m0_basic, trinket_m0, | board_m4
             gemma_m0                      |
--------------------------------------------------------
```
2. Flip the include-o-meter. Uses more vertical space.
```
=================  ===========================================
Board              Modules Available
------------------------------------------------------------
feather_m0_basic | analogio, board, digitalio, pulseio
------------------------------------------------------------
metro_m4_express | analogio, audiobusio, audioio, board
```

If we want to get real crazy, we can look into using something like `plotly` or `matplotlib` and generate raw HTML or an image...

Reply:
I like 2 a lot! Would it be possible to link folks to a particular row and have the modules then link out to their docs? Should we make a separate page for each?

Reply:
> Would it be possible to link folks to a particular row and have the modules then link out to their docs?

I'm sure there's a way. With the Jinja parsing available, might be able to work up some HTML selection box magic. My HTML is lacking; but the internet is vast and full of info. 😄  

> Should we make a separate page for each?

At minimum, it may be best to make the support matrix its own page. With 60+ (and growing) boards, it will be a long list. I'm actually already in WIP-mode to test the second one (first one was easy to mock with existing code).

I also had a thought of expanding this for use on `circuitpython-org`, by providing a function that `build_board_info` can call to place available built-ins for each board. Its good info to have upfront, and can help users select a board based on expected use cases.

Reply:
Example of listing by board:
![support_matrix_by_board](https://user-images.githubusercontent.com/21211479/61919929-55359c80-af1d-11e9-9b22-732711e389f0.png)

Reply:
Nice work! I agree piping this to circuitpython.org would be nice. We could use it in search so we get a grid of all boards with a particular module.

Separate listing page sounds good to me too.